lib/jekyll/document.rb

```diff
@@ -83,7 +83,7 @@ def draft?
     # Returns a String path which represents the relative path
     #   from the site source to this document
     def relative_path
-      @relative_path ||= Pathutil.new(path).relative_path_from(site.source).to_s
+      @relative_path ||= Pathutil.new(path).relative_path_from(focal_point).to_s
```

Comment:
Why can't this be the site.source? Are you thinking the `collections_dir` needs to disappear in the output? Hm. Maybe `relative_path_from(collection.path)`?

Reply:
`collection` does not have an instance method `:path`

Reply:
Any reason it _can't_ have that method? I don't like the `focal_point` method – seems like it's capturing some logic that should be handled somewhere else, like `path`.

Reply:
> Any reason...?

I did not want to **introduce a new `public_method`** in a patch-release..

Reply:
Well now that we have introduce a potential breaking changes in a minor version, we should be above that kind of consideration here. The more `collections_dir` behavior is well thought, the better.

Reply:
@parkr Your call.. should I **introduce a new public_method in a patch-release..**?

Reply:
I think adding Collection#path to fix this bug is a fine solution for a patch release. SemVer makes no assumptions about methods being added, as far as I can tell. Our SemVer is guided by new features (minor bump) and breaking changes (major bump).

Reply:
I would call "enhancing" `Collections` class with a new `public_method` as a feature.. just like removing an existing `public_method` is considered a breaking change..

That said, this is not the place for debating SemVer.. your comment above is clear enough an answer for how to proceed..
Will push a commit soon..